api/src/test/java/org/pmiops/workbench/db/model/StorageEnumsTest.java

```diff
@@ -3,40 +3,68 @@
 import static com.google.common.truth.Truth.assertThat;
 
 import com.google.common.collect.BiMap;
-import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.lang.reflect.ParameterizedType;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.junit.Test;
 
 public class StorageEnumsTest {
-  @Test
-  public void noMissingMapEntries() throws Exception {
-    for (Field f : DbStorageEnums.class.getDeclaredFields()) {
-      if (f.getType() != BiMap.class) {
-        continue;
-      }
 
-      Class enumClass =
-          (Class) ((ParameterizedType) f.getAnnotatedType().getType()).getActualTypeArguments()[0];
+  @Test
+  public void test_noMissingMapEntries() throws Exception {
+    final Set<Class> enumClasses = getEnumerationClasses();
+    final Collection<Method> methods = Arrays.asList(DbStorageEnums.class.getDeclaredMethods());
 
-      Method enumToShort = null;
-      Method shortToEnum = null;
-      for (Method m : DbStorageEnums.class.getDeclaredMethods()) {
-        if (m.getParameterTypes()[0].equals(enumClass)) {
-          enumToShort = m;
-        }
+    // e.g. public static Short reviewStatusToStorage(ReviewStatus s)
+    final Map<Class, Method> toStorageMethods =
+        methods.stream()
+            .filter(m -> enumClasses.contains(firstMethodParameter(m)))
+            .collect(Collectors.toMap(this::firstMethodParameter, m -> m));
 
-        if (m.getReturnType().equals(enumClass)) {
-          shortToEnum = m;
-        }
-      }
+    // e.g. public static ReviewStatus reviewStatusFromStorage(Short s)
+    final Map<Class, Method> fromStorageMethods =
+        methods.stream()
+            .filter(m -> enumClasses.contains(m.getReturnType()))
+            .collect(Collectors.toMap(Method::getReturnType, m -> m));
 
+    for (Class enumClass : enumClasses) {
+      final Method toStorage = toStorageMethods.get(enumClass);
+      final Method fromStorage = fromStorageMethods.get(enumClass);
       for (Object e : enumClass.getEnumConstants()) {
-        Short shortVal = (Short) enumToShort.invoke(null, e);
+        Short shortVal = (Short) toStorage.invoke(null, e);
 
         assertThat(shortVal).named(enumClass.getName() + ":" + e.toString()).isNotNull();
-        assertThat(shortToEnum.invoke(null, shortVal)).isEqualTo(e);
+        assertThat(fromStorage.invoke(null, shortVal)).isEqualTo(e);
```

Comment:
This might be a place where making a local lambda makes things cleaner. E.g. `Function<Short, enumClass> fromStorage = storageValue -> fromStorage.invoke(null, storageValue);` The invoke null takes a while to get used to.